servers/republik/modules/crowdfundings/graphql/resolvers/_mutations/deleteUser.js

```diff
@@ -0,0 +1,184 @@
+  const keepRelations = [ 'pledges', 'stripeCustomers' ]
+  const relations = await pgdb.query(`
+    select
+      R.TABLE_SCHEMA as schema,
+      R.TABLE_NAME as table
+    from INFORMATION_SCHEMA.CONSTRAINT_COLUMN_USAGE u
+    inner join INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS FK
+      on U.CONSTRAINT_CATALOG = FK.UNIQUE_CONSTRAINT_CATALOG
+      and U.CONSTRAINT_SCHEMA = FK.UNIQUE_CONSTRAINT_SCHEMA
+      and U.CONSTRAINT_NAME = FK.UNIQUE_CONSTRAINT_NAME
+    inner join INFORMATION_SCHEMA.KEY_COLUMN_USAGE R
+      on R.CONSTRAINT_CATALOG = FK.CONSTRAINT_CATALOG
+      and R.CONSTRAINT_SCHEMA = FK.CONSTRAINT_SCHEMA
+      and R.CONSTRAINT_NAME = FK.CONSTRAINT_NAME
+    where
+      U.TABLE_NAME = 'users'
+  `)
+    .then(rels => rels
+      .filter(rel => keepRelations.indexOf(rel.table) === -1)
+    )
+  relations.unshift({ // needs to be first
+    schema: 'public',
+    table: 'discussionPreferences'
+  })
+  relations.push({ // needs to be last
+    schema: 'public',
+    table: 'eventLog'
+  })
+  return Promise.all([
+    pgdb.query(`
+      DELETE
+        FROM sessions s
+      WHERE
+        ARRAY[(s.sess #>> '{passport, user}')::uuid] && :userIds
+    `, {
+      userIds: [ userId ]
+    }),
+    ...relations.map(rel =>
+      pgdb[rel.schema][rel.table].delete({
+        userId // assume the foreign column is always called userId
+      })
+    )
+  ])
+}
+
+const getNulledColumnsForUsers = async (pgdb) => {
+  const keepColumns = [ 'firstName', 'lastName', 'addressId', 'createdAt', 'updatedAt' ]
+  return pgdb.queryOneColumn(`
+    SELECT
+      column_name
+    FROM information_schema.columns
+    WHERE
+      table_name = 'users' AND
+      is_nullable = 'YES'
+  `)
+    .then(columns => columns
+      .filter(column => keepColumns.indexOf(column) === -1)
+      .reduce(
+        (acc, column) => ({
+          ...acc,
+          [column]: null
+        }),
+        {}
+      )
+    )
+}
+
+module.exports = async (_, args, context) => {
+  const {
+    userId
+  } = args
+  const {
+    pgdb,
+    req,
+    t,
+    mail: { deleteEmail: deleteFromMailchimp }
+  } = context
+  Roles.ensureUserHasRole(req.user, 'admin')
+
+  const transaction = await pgdb.transactionBegin()
+  try {
+    const user = await transaction.public.users.findOne({
+      id: userId
+    })
+    if (!user) {
+      throw new Error(t('api/users/404'))
+    }
+
+    const pledges = await transaction.public.pledges.find({
+      userId
+    })
+    const memberships = await transaction.query(`
+      SELECT
+        m.*,
+        json_agg(p.*) as pledges
+      FROM memberships m
+      JOIN pledges p
+        ON m."pledgeId" = p.id
+      WHERE
+        m."userId" = :userId
+      GROUP BY
+        m.id
+    `, {
+      userId
+    })
+
+    // returning claimed memberships not supported yet
+    const claimedMemberships = memberships.filter(m => !!m.pledges.find(p => p.userId !== userId))
```

Comment:
I don't understand the question completely. This check throws in the case the user claimed a membership from somebody else, the other way round is not affected. I just didn't come up with a solution in an adequate time, I think in this case we should return the claimed membership to the giver and remember on the membership object, how much of it's time was already consumed. So that when it's claimed again it's just valid for the remaining time. This last part is some work I didn't want to do in this PR.